source/docs/scripts/app.js

```diff
@@ -2,113 +2,217 @@
+
+      .controller('ModalController', ModalController)
+
+      .controller('ModalInstanceControllerOkCancel', ModalInstanceControllerOkCancel)
+
+      .controller('ModalInstanceControllerSelectItem', ModalInstanceControllerSelectItem)
+
+      .controller('ClickToEditController', ClickToEditController);
+
+  function Main(msmNotification) {
+    // use an i18n key here
+    msmNotification.primary('WELCOME');
+  }
+
+  function NotificationController($scope, $log, msmNotification) {
+    $scope.primary = function() {
+      msmNotification.primary('Primary', false);
+    };
+
+    $scope.error = function() {
+      msmNotification.error('Error', false);
+    };
+
+    $scope.success = function() {
+      msmNotification.success('Success', false);
+    };
+
+    $scope.info = function() {
+      msmNotification.info('Info', false);
+    };
+
+    $scope.warning = function() {
+      msmNotification.warning('Warning', false);
+    };
+
+    $scope.clearAll = function() {
+      msmNotification.clearAll();
+    };
+
+    (function initController() {
+      $log.debug('[NotificationController] Initializing...');
+    })();
+  }
+
+  function ModalController($scope, $log, msmModalOkCancel, msmModalSelectFromListing, msmNotification) {
+    var parametersOkCancel = {
+      title: function() {
+        return 'Ok-cancel modal';
+      },
+      text: function () {
+        return 'Here some information text.';
+      }
+    };
+    var parametersSelectItem = {
+      title: function() {
+        return 'Select-from-listing modal';
+      },
+      text: function () {
+        return 'Please select an item:';
+      },
+      items: function() {
+        return [
+          'Item 1',
+          'Item 2',
+          'Item 3'
+        ]
+      }
+    };
+
+    $scope.openOkCancel = function(size) {
+      msmModalOkCancel
+          .open('ModalInstanceControllerOkCancel', parametersOkCancel, size)
+          .result.then(
+          function () {
+            $log.info('Modal (Ok-cancel): Clicked OK.');
+            msmNotification.success('Clicked ok', false);
+          },
+          function () {
+            $log.info('Modal (Ok-cancel): Cancelled.');
+          }
+      );
+    };
+
+    $scope.openSelectItem = function(size) {
+      msmModalSelectFromListing
+          .open('ModalInstanceControllerSelectItem', parametersSelectItem, size)
+          .result.then(
+          function (selectedItem) {
+            $log.info('Modal (select-from-listing): Clicked OK.');
+            msmNotification.success('Selected item: \'' + selectedItem + '\'', false);
+          },
+          function () {
+            $log.info('Modal (select-from-listing): Cancelled.');
+          }
+      );
+    };
+
+    (function initController() {
+      $log.debug('[ModalController] Initializing...');
+    })();
+  }
+
+  function ModalInstanceControllerOkCancel($scope, $log, $modalInstance, title, text) {
+    $scope.title = title;
```

Comment:
We moved to the "Controller as" syntax at Akee. This might be for you as well.
```javascript
.controller('ModalInstanceController', function () {
var ctrl = this;
ctrl.title = title;
// ...
});
```
```html
<section ng-controller="ModalInstanceController as ctrl">
  {{ ctrl.title }}
</section>
```